app/tutorial_servo_control/src/tutorial_servo_control_can.cpp

```diff
@@ -0,0 +1,19 @@
+#include "CANMsg.h"
+#include "rover_config.h"
+#include "mbed.h"
+#include "TutorialServo.h"
+
+int main(){
+    TutorialServo myServo(PA_1);
+    CAN           can(CAN_RX, CAN_TX, ROVER_CANBUS_FREQUENCY);
+    CANMsg        rxMsg;
+
+    while(1){
+        //assume the percentage looks like 0.5
+        if (can.read(rxMsg)){
+            float percentage = rxMsg.data[0];
```

Comment:
This may be where your code is failing to compile: the `data` member is an array of bytes and float here is 4 bytes.

Reply:
hmmm so im a little confused. Can we not store a 1 byte value in a 4 byte data type?

Reply:
You can, but you can't just set them equal. Theres few different ways to do what you are trying to achieve. One is to use a [union](https://en.cppreference.com/w/cpp/language/union), and another would be to type cast a pointer array to a float pointer. This last thing is automated with the `>>` and `<<` operator functions if you use the `CANMsg` class for storing/unpacking messages. You can look at an example of this in the arm or science app `main.cpp` file.